server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -324,6 +327,29 @@ public synchronized void close(final String reason, boolean delete) throws IOExc
         }
     }
 
+    // method is synchronized so that IndexService can't be closed while we're writing out dangling indices information
+    public synchronized void writeDanglingIndicesInfo() {
+        if (closed.get()) {
+            return;
+        }
+        try {
+            IndexMetaData.FORMAT.writeAndCleanup(getMetaData(), nodeEnv.indexPaths(index()));
+        } catch (WriteStateException e) {
+            logger.info(() -> new ParameterizedMessage("failed to write dangling indices state for index {}", index()), e);
```

Comment:
I think a failure here indicates a bad enough problem to warrant a `WARN`-level log.

```suggestion
            logger.warn(() -> new ParameterizedMessage("failed to write dangling indices state for index {}", index()), e);
```

Reply:
3a7b656f5ce